Tags don't get updated to 1password vault

### Your environment

Terraform Provider Version: 1.0.0
Connect Server Version: 1.0.0
OS: macOS Big Sur
Terraform Version: 0.15.1

## What happened?
I'm using the example project provided looking at the code. I see `tags` tags a list(string). I added `tags = ["test"]` to all resources in example project. The apply shows the tags being added but when you look at the state file I see tags being an empty list but apply completed successfully. The resource in vault also don't show any tags either. When I run apply again it tries to add the tags again.

## What did you expect to happen?
If I pass a list of tags it successfully adds the tags 

## Steps to reproduce
1. Add tags to all resources in example project. Tried with one tag or two tags
2. Run plan & apply

```
resource "onepassword_item" "demo_password" {
  vault = var.demo_vault

  title    = "Demo Password Recipe"
  category = "password"

  password_recipe {
    length  = 40
    symbols = false
  }

  tags = ["test"]
}

resource "onepassword_item" "demo_login" {
  vault = var.demo_vault

  title    = "Demo Terraform Login"
  category = "login"
  username = "test@example.com"
  tags = ["test"]
}

resource "onepassword_item" "demo_db" {
  vault    = var.demo_vault
  category = "database"
  type     = "mysql"

  title    = "Demo TF Database"
  username = "root"

  database = "Example MySQL Instance"
  hostname = "localhost"
  port     = 3306
  tags = ["test"]
}

resource "onepassword_item" "demo_sections" {
  vault = var.demo_vault

  title    = "Demo Terraform Item with Sections"
  category = "login"
  username = "test@example.com"

  section {
    label = "Terraform Section"

    field {
      label = "API_KEY"
      type  = "CONCEALED"
      value = "2Federate2!"
    }

    field {
      label = "HOSTNAME"
      value = "example.com"
    }
  }

  section {
    label = "Terraform Second Section"

    field {
      label = "App Specific Password"
      type  = "CONCEALED"

      password_recipe {
        length  = 40
        symbols = false
      }
    }

    field {
      label = "User"
      value = "demo"
    }
  }
  tags = ["test"]
}
```


## Notes & Logs
```
2021-04-29T12:51:43.814-0400 [DEBUG] provider.terraform-provider-onepassword_v1.0.0: 2021/04/29 12:51:43 [ERROR] setting state: Invalid address to set: []string{"notesPlain"}
2021-04-29T12:51:43.815-0400 [WARN]  Provider "provider[\"registry.terraform.io/1password/onepassword\"]" produced an unexpected new value for onepassword_item.demo_db, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .tags: element 0 has vanished
      - .tags: element 1 has vanished
```

--- a/onepassword/data_source_onepassword_item_test.go
+++ b/onepassword/data_source_onepassword_item_test.go
@@ -72,6 +72,7 @@ func compareItemToSource(t *testing.T, dataSourceData *schema.ResourceData, item
 	if dataSourceData.Get("url") != item.URLs[0].URL {
 		t.Errorf("Expected url to be %v got %v", item.URLs[0].URL, dataSourceData.Get("url"))
 	}
+	compareStringSlice(t, getTags(dataSourceData), item.Tags)
 
 	for _, f := range item.Fields {
 		path := f.Label
@@ -158,4 +159,17 @@ func generateFields() []*onepassword.ItemField {
 		},
 	}
 	return fields
+}
+
+func compareStringSlice(t *testing.T, actual, expected []string) {
+	t.Helper()
+	if len(actual) != len(expected) {
+		t.Errorf("Expected slice to be length %d, but got %d", len(expected), len(actual))
+		return
+	}
+	for i, val := range expected {
+		if actual[i] != val {
+			t.Errorf("Expected %s at index %d, but got %s", val, i, actual[i])
+		}
+	}
 }

--- a/onepassword/resource_onepassword_item.go
+++ b/onepassword/resource_onepassword_item.go
@@ -375,6 +375,7 @@ func dataToItem(data *schema.ResourceData) (*onepassword.Item, error) {
 				URL:     data.Get("url").(string),
 			},
 		},
+		Tags: getTags(data),
 	}
 
 	password := data.Get("password").(string)
@@ -582,4 +583,13 @@ func addRecipe(f *onepassword.ItemField, r *onepassword.GeneratorRecipe) {
 		len(f.Value) != r.Length {
 		f.Generate = true
 	}
+}
+
+func getTags(data *schema.ResourceData) []string {
+	tagInterface := data.Get("tags").([]interface{})
+	tags := make([]string, len(tagInterface))
+	for i, tag := range tagInterface {
+		tags[i] = tag.(string)
+	}
+	return tags
 }

--- a/onepassword/resource_onepassword_item_test.go
+++ b/onepassword/resource_onepassword_item_test.go
@@ -24,6 +24,30 @@ var schemaKeys = []string{
 	"tags",
 }
 
+func TestResourceItemToDataDataToTitem(t *testing.T) {
+	resources := map[string]*schema.ResourceData{
+		"login":    generateResourceLoginItem(t),
+		"database": generateResourceDatabaseItem(t),
+		"password": generateResourcePasswordItem(t),
+	}
+	for name, resource := range resources {
+		t.Run(name, func(t *testing.T) {
+			item, err := dataToItem(resource)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			compareItemToSource(t, resource, item)
+
+			convertedResource := schema.TestResourceDataRaw(t, resourceOnepasswordItem().Schema, nil)
+
+			itemToData(item, convertedResource)
+
+			compareResources(t, resource, convertedResource)
+		})
+	}
+}
+
 func TestResourceOnepasswordLoginItemCRUD(t *testing.T) {
 	itemToCreate := generateResourceLoginItem(t)
 	testCRUDForItem(t, itemToCreate)

--- a/onepassword/resource_onepassword_item_test.go
+++ b/onepassword/resource_onepassword_item_test.go
@@ -21,6 +21,7 @@ var schemaKeys = []string{
 	"port",
 	"username",
 	"password",
+	"tags",
 }
 
 func TestResourceOnepasswordLoginItemCRUD(t *testing.T) {
@@ -202,6 +203,7 @@ func generateBaseItem(t *testing.T) *schema.ResourceData {
 	resourceData.Set("vault", "df2e9643-45ad-4ff9-8b98-996f801afa75")
 	resourceData.Set("title", "test_login")
 	resourceData.Set("url", "some_url")
+	resourceData.Set("tags", []string{"tag-1", "tag-2"})
 	return resourceData
 }
 
@@ -214,9 +216,15 @@ func generateResource(t *testing.T, item *onepassword.Item) *schema.ResourceData
 }
 
 func compareResources(t *testing.T, expectedResource, actualResource *schema.ResourceData) {
+	t.Helper()
 	for _, key := range schemaKeys {
-		if expectedResource.Get(key) != actualResource.Get(key) {
-			t.Errorf("Expected %v to be %v but was %v", key, actualResource.Get(key), expectedResource.Get(key))
+		if key == "tags" {
+			compareStringSlice(t, getTags(actualResource), getTags(expectedResource))
+		} else {
+			if expectedResource.Get(key) != actualResource.Get(key) {
+				t.Errorf("Expected %v to be %v but was %v", key, actualResource.Get(key), expectedResource.Get(key))
+			}
 		}
+
 	}
 }

